File: m2-h4/src/main/java/ru/yegorpilipenko/otus/spring/m2h4/service/BookService.java
package ru.yegorpilipenko.otus.spring.m2h4.service;

import ru.yegorpilipenko.otus.spring.m2h4.dto.ShortBook;
import ru.yegorpilipenko.otus.spring.m2h4.model.Book;
import ru.yegorpilipenko.otus.spring.m2h4.model.Comment;

import java.util.Collection;
import java.util.Optional;

public interface BookService {

    Collection<ShortBook> getAll();

    Optional<Book> getById(String id);

    Book save(Book book);

    boolean removeById(String id);

    Book addComment(String bookId, Comment comment);

    Optional<Book> removeComment(String bookId, Comment comment);

}

File: m2-h3/src/main/kotlin/ru/yegorpilipenko/otus/spring/m2h3/service/BookService.kt
package ru.yegorpilipenko.otus.spring.m2h3.service

import ru.yegorpilipenko.otus.spring.m2h3.dto.ShortBook
import ru.yegorpilipenko.otus.spring.m2h3.model.Book
import ru.yegorpilipenko.otus.spring.m2h3.model.Comment

interface BookService {

    fun getAll(): Collection<ShortBook>

    fun getById(id: String): Book?

    fun save(book: Book): Book

    fun removeById(id: String): Boolean

    fun addComment(bookId: String, comment: Comment): Book

    fun removeComment(bookId: String, comment: Comment): Book?

}
